dask_planner/src/parser.rs

```diff
@@ -0,0 +1,183 @@
+        let mut tokenizer = Tokenizer::new(dialect, sql);
+        let tokens = tokenizer.tokenize()?;
+
+        Ok(DaskParser {
+            parser: Parser::new(tokens, dialect),
+        })
+    }
+
+    #[allow(dead_code)]
+    /// Parse a SQL statement and produce a set of statements with dialect
+    pub fn parse_sql(sql: &str) -> Result<VecDeque<DaskStatement>, ParserError> {
+        let dialect = &DaskDialect {};
+        DaskParser::parse_sql_with_dialect(sql, dialect)
+    }
+
+    /// Parse a SQL statement and produce a set of statements
+    pub fn parse_sql_with_dialect(
+        sql: &str,
+        dialect: &dyn Dialect,
+    ) -> Result<VecDeque<DaskStatement>, ParserError> {
+        let mut parser = DaskParser::new_with_dialect(sql, dialect)?;
+        let mut stmts = VecDeque::new();
+        let mut expecting_statement_delimiter = false;
+        loop {
+            // ignore empty statements (between successive statement delimiters)
+            while parser.parser.consume_token(&Token::SemiColon) {
+                expecting_statement_delimiter = false;
+            }
+
+            if parser.parser.peek_token() == Token::EOF {
+                break;
+            }
+            if expecting_statement_delimiter {
+                return parser.expected("end of statement", parser.parser.peek_token());
+            }
+
+            let statement = parser.parse_statement()?;
+            stmts.push_back(statement);
+            expecting_statement_delimiter = true;
+        }
+        Ok(stmts)
+    }
+
+    /// Report unexpected token
+    fn expected<T>(&self, expected: &str, found: Token) -> Result<T, ParserError> {
+        parser_err!(format!("Expected {}, found: {}", expected, found))
+    }
+
+    /// Parse a new expression
+    pub fn parse_statement(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.peek_token() {
+            Token::Word(w) => {
+                match w.keyword {
+                    Keyword::CREATE => {
+                        // move one token forward
+                        self.parser.next_token();
+                        // use custom parsing
+                        self.parse_create()
+                    }
+                    _ => {
+                        // use the native parser
+                        Ok(DaskStatement::Statement(Box::from(
+                            self.parser.parse_statement()?,
+                        )))
+                    }
+                }
+            }
+            _ => {
+                // use the native parser
+                Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_statement()?,
+                )))
+            }
+        }
+    }
+
+    /// Parse a SQL CREATE statement
+    pub fn parse_create(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.parse_identifier() {
```

Comment:
we could use the `parse_keyword` method here since `create` is a recognized keyword in sqlparser. 

More generally if this was not recognized as a keyword by sqlparser, would we have to create our own keyword list in dask-sql?

Reply:
As comment ^^ suggests I had just made a typo. Should be "model" not "create". "model" is not a `Keyword` so we use Ident instead.